exchange/replication_test.go

```diff
@@ -499,6 +499,34 @@ func TestMultiDispatchStreams(t *testing.T) {
 		tnds[tnode.Host.ID()] = tnode
 	}
 
+	// Create a provider that already has the file
+	tnode := testutil.NewTestNode(mn, t)
+	tnode.SetupDataTransfer(ctx, t)
+	t.Cleanup(func() {
+		err := tnode.Dt.Stop(ctx)
+		require.NoError(t, err)
+	})
+	tnode.LoadFileToNewStore(ctx, t, fname)
+	idx2, err := NewIndex(tnode.Ds)
+	require.NoError(t, err)
+	opts2 := Options{Regions: regions, MultiStore: tnode.Ms, Blockstore: tnode.Bs}
+	hn1, err := NewReplication(tnode.Host, idx, tnode.Dt, NewMockRetriever(tnode.Dt, idx2), opts2)
+	require.NoError(t, err)
+	ref := &DataRef{
+		PayloadCID:  rootCid,
+		PayloadSize: int64(256000),
+	}
+
+	// the file is already in the node so setRef should return ErrRefAlreadyExists
+	require.Error(t, idx.SetRef(ref), ErrRefAlreadyExists)
+
+	// we update the ref instead of adding it
+	require.NoError(t, idx.UpdateRef(ref))
```

Comment:
Not sure what you're testing here. You probably want to add a separate test in the `index_test.go` that specifically verifies that the keys are properly merged when updating a ref.